### PR TITLE
Device API review

### PR DESCRIPTION
**Pull request by BorjaRequena**
Reviewing the PRs for the new device API, I am finding a few minor things that require fixing. I am gathering them in this PR to fix them before the release.

### CODE REVIEW

File: pennylane/measurements/counts.py [hunk 1]
```diff
@@ -56,6 +56,12 @@ def counts(op=None, wires=None, all_outcomes=False) -> "CountsMP":
     :math:`p(\lambda_i) = |\langle \xi_i | \psi \rangle|^2`, where :math:`| \xi_i \rangle`
     is the corresponding basis state from the observable's eigenbasis.
 
+    .. note::
+
+        QNodes that return counts cannot, in general, be differentiated, since the derivative
+        with respect to a stochastic process is ill-defined. Please, refer to
+        :func:`~.pennylane.sample` if differentiability is required.
+
```
Comment (eddddddy) [inline pennylane/measurements/counts.py:64]: We definitely _could_ support counts if we really wanted to, see [here](https://github.com/PennyLaneAI/pennylane/issues/2932). There's nothing fundamental that's stopping us, other than some (probably) ugly technical details.

Maybe something like this?
```suggestion
    .. note::

        Differentiation of QNodes that return ``counts`` is currently not supported. Please refer to
        :func:`~.pennylane.sample` if differentiability is required.

```

**Review by mudit2812 — APPROVED** [review]
I like @eddddddy 's suggestion, but otherwise, this looks good!

File: pennylane/measurements/counts.py (hunk 1 above)
Comment (BorjaRequena) [reply pennylane/measurements/counts.py:64]: Alright. I like it!

**Review by eddddddy — APPROVED** [review]
Great work @BorjaRequena 🎉

File: pennylane/measurements/counts.py [hunk 2]
```diff
@@ -56,6 +56,11 @@ def counts(op=None, wires=None, all_outcomes=False) -> "CountsMP":
     :math:`p(\lambda_i) = |\langle \xi_i | \psi \rangle|^2`, where :math:`| \xi_i \rangle`
     is the corresponding basis state from the observable's eigenbasis.
 
+    .. note::
+
+        Differentiation of QNodes that return ``counts`` is currently not supported. Please refer to
+        :func:`~.pennylane.sample` if differentiability is required.
+
```
Comment (albi3ro) [inline pennylane/measurements/counts.py:63]: Pretty sure you can't differentiate with respect to counts either.

Comment (albi3ro) [reply pennylane/measurements/counts.py:63]: Oh I see the comment below. Single shot expectation value.

Comment (BorjaRequena) [reply pennylane/measurements/counts.py:63]: Should we then go back to how it was before [this update](https://github.com/PennyLaneAI/pennylane/pull/4283#discussion_r1239029103)?

**Pull request merged**